### ISSUE TITLE
No option to add search plugin with customized preferences

### ISSUE THREAD

**Issue by Poorchop** [searx/searx#1251]
The about page provides a method to add searx to Firefox as a custom search engine, but it doesn't retain any user customizations. StartPage provides a method for adding their search engine to Firefox with custom parameters. Like searx, StartPage provides a custom URL with custom settings embedded, and it uses this as the basis for adding the custom search plugin. Searx should provide the same so that settings like safesearch and locale are preserved even after deleting cookies.

**Comment by cloo** [comment]
Agree. I've got a self-hosted searx engine, but it is incovenient, while I can't find information how to make it default on firefox, because plugin only points to searx.me.

**Comment by asciimoo** [comment]
@cloo browsers use the `ShortName` attribute in `opensearch.xml` to identify search engines. Change the `instance_name` attribute in `settings.yml` and restart your searx and reload your browser, then you should see your instance.

**Comment by cloo** [comment]
I have deleted my comment before saw yours, because I actually found a way to do it nicely. 
(For the record - I complained that there is no way to add searx as search engine in firefox)
Solution is simpler - for some reason just have to switch on the search bar (instead of using unified search/address bar) and then you can add any custom search engine.
Maybe this instruction could be added to docs?

**Comment by Pofilo** [comment]
> for some reason just have to switch on the search bar (instead of using unified search/address bar) and then you can add any custom search engine.

Yes this solution works but it may change in the future so adding this to docs seems a bit overrated. It's a Firefox UI problem !

**Comment by cloo** [comment]
True, but as (hopefully) most users, at first I want to know about UI, and if possible, avoid understanding internals.
Don't know your docs policy, so can't argue what fits in and what doesnt. 
If it does, one could also suggest to describe procedures in other popular browsers as well. So just my two cents at this point.

**Comment by unixfox** [comment]
Related to https://github.com/searx/searx/issues/2082
